runtime/src/commonMain/kotlin/pbandk/internal/json/JsonValueMarshaller.kt

```diff
@@ -105,4 +106,23 @@ internal class JsonValueMarshaller(private val jsonConfig: JsonConfig) {
             writeValue(k, keyType).content to writeValue(v, valueType)
         }
     }
+
+    private fun Value.toElement(): JsonElement = when (this.kind) {
+        is Value.Kind.StringValue -> writeString(this.stringValue!!)
+        is Value.Kind.BoolValue -> writeBool(this.boolValue!!)
+        is Value.Kind.NumberValue -> writeDouble(this.numberValue!!)
+        is Value.Kind.StructValue -> writeStruct(this.structValue!!)
+        is Value.Kind.ListValue -> jsonArray {
```

Comment:
🔧 In addition to moving the handling of `ListValue` up to the top-level `when` statement in `writeMessage()` (similar to my suggestion for `Value`), you should have it delegate to the existing `writeValue()` method (which will delegate to `writeRepeated()`) in order to write out the array instead of repeating the same `jsonArray` code here.